### ISSUE TITLE
Improve Delete Site UI with confirmation mechanics

### ISSUE THREAD

**Issue by sumerjabri** [craftercms/craftercms#6261]
### Latest version

- [X] The issue is in the develop branch
- [ ] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Duplicates

- [X] I have searched the existing issues

### Is your enhancement request related to a problem? Please describe.

Deleting a Site is irreversible and should be made clear to the user. Additionally, there needs to be a small speedbump to require the user to read and understand the consequences of their action.

### Describe the solution you'd like

Add a Delete Site confirmation dialog explaining to the user that this action cannot be undone and requires them to enter the Site Identifier before deleting the site. Ping me for an example.

**Comment by rart** [comment]
@jvega190 change the dropdown to be a confirm-style dialog with a checkbox alert similar to the delete dialog state the action is irreversible. Finally, the site card should reflect a disabled state while it is being deleted, before it gets removed from the list.

```
Confirm the permanent deletion of the "Editorial" project.

[ ] I understand deleting a project is immediate and irreversible 

[ Delete ]
[ Cancel ] 
```

**Comment by yacdaniel** [comment]
![image](https://github.com/user-attachments/assets/eda06da4-5c57-4672-8df3-42a6b92aab21)

Verified fixed in `4.2.0-SNAPSHOT`
